Source/WebGPU/WGSL/Lexer.cpp

```diff
@@ -1024,8 +1025,17 @@ Token Lexer<T>::lexNumber()
         return convert(result);
     }
 
+    auto length = static_cast<size_t>(m_code.position() - start.data());
+    if (suffix)
+        length--;
+    auto buffer = MallocSpan<char>::malloc(length + 1);
```

Comment:
Having to malloc and copy is so annoying!

But I don't see an obvious better way that's easy.

Maybe you can pick a maximum number of characters that a float can ever be, and copy into an std::array on the stack, so at least you don't have to separately malloc, even though you do have to copy.

Reply:
I did consider either that or `calloc`ing, but I guess having a maximum length is reasonable enough, I'll refactor it.

Reply:
I ended up making a Vector with a large inline capacity, so we avoid the malloc in the common case but still support hex floats of any length.